src/v3/src/transformer/phone/phoneEnrollmentTransformer.ts

```diff
@@ -76,37 +77,62 @@ export const transformPhoneEnrollment: IdxStepTransformer = ({ formBag, transact
     },
   };
 
-  const stepper: StepperLayout = {
-    type: UISchemaLayoutType.STEPPER,
-    elements: [
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          smsInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          smsStepSubmitButton,
-        ],
-      },
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          voiceInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          voiceStepSubmitButton,
-        ],
-      },
-    ],
+  const phoneOptionLayouts: { [key:string]: UISchemaLayout | StepperLayout } = {
+    sms: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        smsInfoTextElement,
+        phoneNumberElement,
+        smsStepSubmitButton,
+      ],
+    },
+    voice: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        voiceInfoTextElement,
+        phoneNumberElement,
+        voiceStepSubmitButton,
+      ],
+    },
+    both: {
+      type: UISchemaLayoutType.STEPPER,
+      elements: [
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            smsInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            smsStepSubmitButton,
+          ],
+        },
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            voiceInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            voiceStepSubmitButton,
+          ],
+        },
+      ],
+    } as StepperLayout,
   };
 
   uischema.elements = [
     titleElement,
-    stepper,
   ];
 
+  const phoneMethodOptions = methodTypeElement!.options!.inputMeta.options!;
+  if (phoneMethodOptions.length === 1) {
+    const methodType = phoneMethodOptions[0].value.toString();
+    uischema.elements.push(phoneOptionLayouts[methodType]);
+  } else {
+    uischema.elements.push(phoneOptionLayouts.both);
+  }
+
   // set default dataSchema
-  dataSchema.submit = smsStepSubmitButton.options;
+  dataSchema.submit = phoneMethodOptions[0].value === 'sms' ? smsStepSubmitButton.options : voiceStepSubmitButton.options;
```

Comment:
Since we wont add action params, you can revert this (since the options to both buttons are the same)